src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs

```diff
@@ -318,7 +318,7 @@ private static void ApplyStyleOnWindow(Window window, bool useLightColors)
 
     #region Internal Properties
 
-    internal static bool DeferSyncingThemeModeAndResources { get; set; } = true;
+    internal static bool DeferSyncingThemeModeAndResources { get; set; } = Standard.Utility.IsOSWindows10OrNewer ? true : false;
```

Comment:
Could just be `internal static bool DeferSyncingThemeModeAndResources { get; set; } = Standard.Utility.IsOSWindows10OrNewer`.

But do we need this at all? .NET 8 already doesn't run on less than Win8: https://github.com/dotnet/core/blob/main/release-notes/8.0/supported-os.md

Reply:
Since the call that makes this flag `False` occurs from inside an `if` block with this OS version check, I didn't want to leave  this `True` for other cases.

Reply:
I understand but is there any reason we should be checking if the Windows build is less than `10240` altogether? The runtime shouldn't even install in that case and is certainly unsupported, the lowest supported is `14393` (1607).

And since less than Win10 is not supported by the runtime, then I see no reason for the checks altogether.
I understand this was previously Win11 check and it was just modified but do we need it? Not in my opinion.

Reply:
Yes, exactly, two different things.

If you think we should keep those checks around for unsupported system when it's a new function, how crazy should we get? Do we implement every new feature with checks for unsupported systems and then try to figure out some sort of fallback if it ain't supported? I'm obviously exaggerating but you get my point.

Even JIT has now removed tons of Win95 compat code for .NET 9, but I can still select compat mode for Win95 on latest Win10, guess I might have some issues when I'm lying in `PEB` to `RtlGetVersion`.

Reply:
I don't know the answer. I have never heard from anyone in .NET about how supported running under compatibility mode is or what the expectations are. Just pointing out that user can relatively easily run the application in a way that the check makes a difference.

Reply:
I agree with @miloush on this one, we have seen customers use applications in ways that I didn't expect before. Moreover, if the time to purge has come, then we will do it as a whole.

Reply:
Fine by me, was just really looking for opinions on this.

Reply:
@h3xds1nz , I agree with you that there are definitely checks in the codebase which won't be triggered, but at the same time, I also think that we should deal with them at once.

Reply:
Yeah, agreed, this can be looked at later with a complex PR. Thank you for the opinion.